src/creature_states_combt.c

```diff
@@ -3124,7 +3124,7 @@ TbBool creature_look_for_enemy_heart_combat(struct Thing *thing)
         struct CreatureControl* cctrl = creature_control_get_from_thing(thing);
         heartng = thing_get(cctrl->combat.battle_enemy_idx);
         if (thing_is_dungeon_heart(heartng)) {
-            return false;
+            return true;
```

Comment:
This will affect line 2732, which is in creature_in_combat_wait().
When we're in creature_in_combat_wait(), I suppose the creature is waiting to fight a thing but is not currently attacking.
So, returning false gives the creature a chance to switch to a better enemy. Returning true will make the creature keep waiting.
I am new to this project, please correct me if I am wrong.